traversal/scanner/GraphIterator.java

```diff
@@ -377,18 +377,70 @@ private void reset() {
         }
 
         private Forwardable<Vertex<?, ?>, Order.Asc> createIteratorFromEdges() {
-            List<Forwardable<Vertex<?, ?>, Order.Asc>> iters = new ArrayList<>();
-            edgeInputs.forEach((procedureEdge, vertex) -> iters.add(branch(vertex, procedureEdge)));
-            if (iters.size() == 1) return iters.get(0);
-            else return intersect(iterate(iters), ASC);
+            Map<Vertex<?, ?>, List<ProcedureEdge.Native.Thing.RolePlayer>> overlappingRPEdges = new HashMap<>();
+            List<ProcedureEdge<?, ?>> otherEdges = new ArrayList<>();
```

Comment:
to handle parallel/overlapping role player edges from the same vertex, we intersect them separately to make sure the roles they occupy are valid.